script.js

```diff
@@ -0,0 +1,91 @@
+let books = [];
+const Book = {
+  id: null,
+  title: '',
+  author: '',
+};
+
+const addBtn = document.querySelector('#addBtn');
+const bookTitle = document.querySelector('#book-title');
+const bookAuthor = document.querySelector('#book-author');
+const bookList = document.querySelector('#book-list');
+
+const getLocalStorageBooks = () => JSON.parse(localStorage.getItem('books'))[0];
+
+const getLocalIdBook = () => JSON.parse(localStorage.getItem('books'))[1];
+
+const createLocalStorage = (id) => {
+  const init = JSON.stringify([[], id]);
+  localStorage.setItem('books', init);
+};
+
+const createBookId = () => {
+  getLocalStorageBooks();
+  const bookID = getLocalIdBook();
+  if (!bookID) return 1;
+  return bookID + 1;
+};
+
+const removeItem = (id) => {
+  const newArray = books.filter((item) => item.id !== id);
+  const currentStateId = getLocalIdBook();
+  const objectString = JSON.stringify([newArray, currentStateId]);
+  localStorage.setItem('books', objectString);
+};
+
+const createBookListItem = () => {
+  if (books === null) {
+    createLocalStorage(null);
+  }
+
+  books = getLocalStorageBooks();
+
+  bookList.innerHTML = '';
+
+  books.forEach((element) => {
+    const bookItem = document.createElement('li');
+    bookItem.id = element.id;
+
+    const bookTitleEl = document.createElement('p');
+    const bookAuthor = document.createElement('p');
+
+    const removeBtn = document.createElement('button');
+    removeBtn.id = element.id;
+    removeBtn.classList.add('remove-btn');
+
+    removeBtn.textContent = 'Remove';
+
+    removeBtn.addEventListener('click', () => {
+      removeItem(Number(removeBtn.id));
+      createBookListItem();
+    });
+
+    bookTitleEl.textContent = element.title;
+    bookAuthor.textContent = element.author;
+
+    bookItem.appendChild(bookTitleEl);
+    bookItem.appendChild(bookAuthor);
+    bookItem.appendChild(removeBtn);
+
+    bookList.appendChild(bookItem);
+  });
+};
+
+addBtn.addEventListener('click', (event) => {
+  event.preventDefault();
+
+  Book.id = createBookId();
+  Book.title = bookTitle.value;
+  Book.author = bookAuthor.value;
+
+  books = getLocalStorageBooks();
+
+  books.push(Book);
+
+  const booksString = JSON.stringify([books, Book.id]);
+  localStorage.setItem('books', booksString); // updating the books array
+
+  createBookListItem();
+});
+
```

Comment:
Good job setting up the functionalities, however, the add functionality is not working currently and I think you should fix the errors popping up on the console once the button is clicked in order for the functions to be accessed by a user.

[screen-capture (1).webm](https://github.com/tajulafreen/Awesome-Books/assets/84930355/3e1dec72-0beb-46a4-baf2-163d192422b3)

Reply:
Hello @DuaneDave, 🙇‍♂️ Thank you for the code review, We finally fixed the issue for add button